runtime/doc/options.txt

```diff
@@ -422,8 +422,13 @@ command, not when assigning a value to an option with ":let".
 
 							*$HOME-windows*
 On MS-Windows, if $HOME is not defined as an environment variable, then
-at runtime Vim will set it to the expansion of $HOMEDRIVE$HOMEPATH.
-
+at runtime Vim will set it to the expansion of $HOMEDRIVE$HOMEPATH. If
+$HOMEDRIVE and $HOMEPATH is not set, $USERPROFILE is used. This is an internal
```

Comment:
The test

```
 +    UnletEnv $HOME
 +    let $USERPROFILE = 'C:\foo'
 +    let $HOMEDRIVE = 'C:'
 +    let $HOMEPATH = '\baz'
 +    call CheckHome('C:\foo')
```

checks that `$USERPROFILE` has priority over `$HOMEDRIVE$HOMEPATH`. Though given that first two lines of CheckHome were commented I can’t say whether this is true based on the test.

Also test

```
 +    UnletEnv $HOME $USERPROFILE $HOMEPATH
 +    let $HOMEDRIVE = 'C:'
 +    call CheckHome('C:\')
```

checks that `$HOMEDRIVE` is enough to get `$HOME`, so “if $HOMEDRIVE and $HOMEPATH is not set” is not true (and also have a typo), need “if $HOMEDRIVE is not set”. Same note about tests not checking anything due to commented out lines applies.